packages/react-mutation-mapper/src/component/track/TrackRect.tsx

```diff
@@ -0,0 +1,146 @@
+import * as React from 'react';
+import { action, computed, makeObservable, observable } from 'mobx';
+import { observer } from 'mobx-react';
+
+type TrackRectProps = {
+    x: number;
+    y: number;
+    width?: number;
+    height?: number;
+    isHovered: boolean;
+    hitZoneClassName?: string;
+    hitZoneXOffset?: number;
+    spec: TrackRectSpec;
+};
+
+export type TrackRectSpec = {
+    startCodon: number;
+    endCodon?: number;
+    label?: string;
+    labelColor?: string;
+    color?: string;
+    tooltip?: JSX.Element;
+};
+
+@observer
+export default class TrackRect extends React.Component<TrackRectProps, {}> {
+    @observable private textElt: SVGTextElement | null = null;
+    private handlers: any;
+    constructor(props: any) {
+        super(props);
+        makeObservable(this);
+        this.state = {
+            displayText: props.label || '',
+        };
```

Comment:
We should not use react's built-in state. Do we really need a `displayText` here? I see another `@computed get displayText()` and I think that's the only one we are using.